client/utils/utils.go

```diff
@@ -69,7 +69,11 @@ func CompleteAndBroadcastTxCLI(txBldr authtxb.TxBuilder, cliCtx context.CLIConte
 			return err
 		}
 
-		fmt.Fprintf(os.Stderr, "%s\n\n", cliCtx.Codec.MustMarshalJSON(stdSignMsg))
+		json, err := cliCtx.Codec.MarshalJSONIndent(stdSignMsg, "", "  ")
```

Comment:
Don't we have a flag for enabling/disabling indentation?

Reply:
We do -- I wonder if we should use it here? I'm indifferent here either way as it's just a prompt.

Reply:
The difference between enabling/disabling indentation is the size of json string and readability. The size is important only while saving origin json to file. The readability is very important for people using cli. 

The reality is that CLI is the only way for human interact with chain. And I can't imagine a scenario where the CLI output json is directly saved to a text file.

So, maybe we can even consider removing the flag `--indent` to reduce the developer's workload and the user's input workload.

Reply:
I do not think we should remove `--indent` flag. The CLI output is saved to disk in many instances such as generate only.

Reply:
I think CLI has two scenes:
1. **Human interact with chain**. They want to understand output as soon as possible.
2. **Automatic test**. First, the output is used for result comparing by computer. Finally, human will find the difference between expect output and actual output on failure case. And the report will be deleted after few days.

I haven't found the scenes that CLI's stdout/stderr need to be compressed saved for a long time. 

Would you give me some examples? Thanks

Reply:
@yangyanqing JSON is our machine-parseable format, and because of that I in line of principle agree with you that `--indent` is superfluous. Though in those cases where we rely on JSON as human readable format too, `--indent` comes handy.

BTW, I'd replace `Automatic test` with `Machine interaction`. Let's not forget that our command line utility is used by third-party scripts, that's why we always should tread carefully when it comes to ABI changes.

Reply:
OK, I will add the flag --indent. :)

Reply:
To be clear - I think indentation by default is fine, we should just have a flag to disable it then.

Reply:
What is `--indent` current default? Is it indentation on or off by default?

Reply:
Whatever it is at the moment, I think we should default to enabling indentation.

Reply:
We can set `--indent` true default through `gaiacli config indent true`.